crates/relayer/src/worker/channel.rs

```diff
@@ -19,6 +19,7 @@ pub fn spawn_channel_worker<ChainA: ChainHandle, ChainB: ChainHandle>(
     chains: ChainHandlePair<ChainA, ChainB>,
     cmd_rx: Receiver<WorkerCmd>,
 ) -> TaskHandle {
+    let mut clear_pending = true;
```

Comment:
Would `complete_handshake_on_new_block` be a better name for this, since we are not actually clearing packets in the worker?

Reply:
we are clearing a pending (vs live) handshake event. but i also like your proposal, will change.